### ISSUE TITLE
Detect outdated dependencies in pom.xml

### ISSUE THREAD

**Issue by DavyLandman** [usethesource/rascal-language-servers#523]
That should have given a user a warning that would have made #520 and #521 a lot nicer.

(something along the lines of, hey please update to a minimal version of rascal 0.40.17)

**Comment by jurgenvinju** [comment]
This is already implemented on the no-more-libs PR.

**Comment by jurgenvinju** [comment]
https://github.com/usethesource/rascal/pull/1969

The new messages field in PathConfig will contain exactly those messages you proposed for the Rascal project and for rascal-lsp.

Also self-application is detected, which influences the library path and yieldd an info message. 

This feature does not use exact error locations yet. I thought we can wait until the standard library is bootstrapped and we can use the XML origins for that. For now we report issues on the first live of the pom.xml or .MF file.

**Comment by jurgenvinju** [comment]
What I don't understand yet is why the typechecker didn't detect the old .tpl files as being of the wrong version in #520. Is the old typechecker somehow in play ?

**Comment by DavyLandman** [comment]
no, @PaulKlint suspects this was a case he hadn't covered yet in the detection of this error. he would pick this up soon.

**Comment by toinehartman** [comment]
Related: https://github.com/usethesource/rascal-language-servers/issues/586

**Comment by toinehartman** [comment]
To test:
- Old LSP in POM, new LSP from extension: what happens?
